src/libServer/LookupServer.cpp

```diff
@@ -1182,16 +1182,23 @@ Json::Value LookupServer::GetEthBlockCommon(const TxBlock& txBlock,
                                                  includeFullTransactions);
 }
 
-Json::Value LookupServer::GetEthBalance(const std::string& address) {
-  const auto balanceStr = this->GetBalance(address, true)["balance"].asString();
+Json::Value LookupServer::GetEthBalance(const std::string& address,
+                                        const std::string& tag) {
+  if (tag == "latest" || tag == "earliest" || tag == "pending") {
```

Comment:
Let's mark it with a TODO and file a ticket saying that this is a limited current implementation and we'll need to have a full implementation of this for pending and earliest and any other block.

Reply:
https://zilliqa-jira.atlassian.net/browse/ZIL-4771